src/gui/transferlistwidget.cpp

```diff
@@ -757,6 +758,57 @@ void TransferListWidget::editTorrentTrackers()
     trackerDialog->open();
 }
 
+void TransferListWidget::exportTorrent()
+{
+    if (getSelectedTorrents().isEmpty())
+        return;
+
+    auto fileDialog = new QFileDialog(this, tr("Choose folder to save exported .torrent files"));
+    fileDialog->setAttribute(Qt::WA_DeleteOnClose);
+    fileDialog->setFileMode(QFileDialog::Directory);
+    fileDialog->setOptions(QFileDialog::ShowDirsOnly);
+    connect(fileDialog, &QFileDialog::fileSelected, this, [this](const QString &dir)
+    {
+        const QVector<BitTorrent::Torrent *> torrents = getSelectedTorrents();
+        if (torrents.isEmpty())
+            return;
+
+        const Path savePath {dir};
+        if (!savePath.exists())
+            return;
+
+        const QString errorMsg = tr("Export .torrent file failed. Torrent: \"%1\". Save path: \"%2\". Reason: \"%3\"");
+
+        bool hasError = false;
+        for (const BitTorrent::Torrent *torrent : torrents)
```

Comment:
The main problem with all such batch actions is that they block the main thread, which leads to GUI hanging. Apparently, we should think about how to fix it.

Reply:
>Apparently, we should think about how to fix it.

What about using a thread to handle this loop? Note that this will access `BitTorrent::Torrent` from a different thread, I hope that won't be a problem..

Reply:
> I hope that won't be a problem..

Torrent can be deleted.

I don't ask you to solve it in this PR. It can be harder than it looks at a first glance.